packages/block-editor/package.json

```diff
@@ -52,6 +52,7 @@
 		"@wordpress/keyboard-shortcuts": "file:../keyboard-shortcuts",
 		"@wordpress/keycodes": "file:../keycodes",
 		"@wordpress/notices": "file:../notices",
+		"@wordpress/preferences": "file:../preferences",
```

Comment:
I'm not really sure that I like that personally, I feel the block-editor package shouldn't be depending on the "preferences" package. The reason is that the "preferences" package is something that is generic now but I see it as more tied to WP especially if we implement later API based preferences (persist in the user profile)

I guess we can implement this WP relationship in a non-intrusive way, using adapters or something but maybe we should cross that bridge at that moment?

Reply:
Yeah, this is something I was unsure about.

There's also #39632 where I've migrated some of the block-editor's persisted data to the preferences package, that would also add the dependency.

Reply:
In https://github.com/WordPress/gutenberg/pull/39632 it seems it's even harder to remove that dependency. So maybe we can just accept it as a dependency but make sure that any "WP" dependency should be added in an optional "adapter" layer

Reply:
You can also look at the Block Directory integration:
https://github.com/WordPress/gutenberg/tree/trunk/packages/block-directory

It's shaped as an optional plugin that integrates with the block editor. Although, I'm not sure how that would work with preferences, to be honest.

Reply:
The block directory renders into slots IIRC, so it seems a bit easier to make optional.

As Riad mentions, https://github.com/WordPress/gutenberg/pull/39632 (block insert usage, which is used to power the quick inserter) is a more difficult thing to make optional.

There's no hurry to merge this or #39632, so it could be worth exploring how a flexible persistence layer in preferences might be integrated first.

Reply:
Yes, the block directory integration was easier to make optional. It's worth also pursuing the same approach because it makes the editor easier to extend assuming you would find a decent way to orchestrate the preferences store. It would benefit also plugin authors. I'm happy to pair with you to discuss the challenges and possible ideas.

Reply:
There's an experiment here to try making the persistence aspect of preferences optional/configurable - #39795.